examples/dpo_demo_gemma3.ipynb

```diff
@@ -0,0 +1,849 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -q kagglehub\n",
+    "\n",
+    "!pip install -q tensorflow\n",
+    "!pip install -q tensorboardX\n",
+    "!pip install -q grain\n",
+    "!pip install -q git+https://github.com/google/tunix\n",
+    "!pip install -q git+https://github.com/google/qwix\n",
+    "\n",
+    "!pip uninstall -q -y flax\n",
+    "!pip install -q git+https://github.com/google/flax.git\n",
+    "\n",
+    "!pip install -q huggingface_hub\n",
+    "!pip install -q datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import functools\n",
+    "import os\n",
+    "import re\n",
+    "\n",
+    "from flax import nnx\n",
+    "import grain\n",
+    "import humanize\n",
+    "import jax\n",
+    "import optax\n",
+    "import qwix\n",
+    "import tensorflow_datasets as tfds\n",
+    "from tqdm.auto import tqdm\n",
+    "\n",
+    "from tunix.examples.data import translation_dataset as data_lib\n",
+    "from tunix.generate import sampler as sampler_lib\n",
+    "from tunix.models.gemma3 import params_safetensors as params_safetensors_lib\n",
+    "\n",
+    "from datasets import load_dataset\n",
+    "from tunix.sft.dpo.dpo_trainer import DpoTrainingConfig\n",
+    "from tunix.sft.dpo.dpo_trainer import DpoTrainer\n",
+    "from tunix.sft.dpo.dpo_trainer import TrainingInput\n",
+    "from huggingface_hub import snapshot_download\n",
+    "from tunix.sft.dpo.dpo_trainer import _generate_ids_and_masks\n",
```

Comment:
Do we need this? It's a private function. Can we use `process_dpo_record` directly?